File: src/Application Framework/Validation/BaseValidator.cs
using Microsoft.VisualBasic;
using System;

namespace FSM
{
    public class BaseValidator
    {
        
        protected ValidatorMessages _validatorMessages;

        public ValidatorMessages ValidatorMessages
        {
            get
            {
                return _validatorMessages;
            }

            set
            {
                _validatorMessages = value;
            }
        }

        public string CriticalMessagesString()
        {
            string msgString = "";
            foreach (string s in _validatorMessages.CriticalMessages)
                msgString += s + Constants.vbCrLf;
            return msgString;
        }

        public string WarningMessageString()
        {
            string msgString = "";
            foreach (string s in _validatorMessages.WarningMessages)
                msgString += s + Constants.vbCrLf;
            return msgString;
        }

        
        

        public BaseValidator()
        {
            _validatorMessages = new ValidatorMessages();
        }

        public void AddCriticalMessage(string message)
        {
            _validatorMessages.CriticalMessages.Add(message);
        }

        public void AddWarningMessage(string message)
        {
            _validatorMessages.WarningMessages.Add(message);
        }

        
    }

    public class ValidationException : Exception
    {
        
        private BaseValidator m_valid = new BaseValidator();

        public BaseValidator Validator
        {
            get
            {
                return m_valid;
            }

            set
            {
                m_valid = value;
            }
        }

        
        

        public ValidationException(BaseValidator inValidator)
        {
            m_valid = inValidator;
        }

        
    }
}

File: src/Application Framework/Validation/BaseValidator.vb
Public Class BaseValidator

#Region "Properties"

    Protected _validatorMessages As ValidatorMessages

    Public Property ValidatorMessages() As ValidatorMessages
        Get
            Return Me._validatorMessages
        End Get
        Set(ByVal value As ValidatorMessages)
            Me._validatorMessages = value
        End Set
    End Property

    Public Function CriticalMessagesString() As String
        Dim msgString As String = ""
        For Each s As String In _validatorMessages.CriticalMessages
            msgString += s & vbCrLf
        Next
        Return msgString
    End Function

    Public Function WarningMessageString() As String
        Dim msgString As String = ""
        For Each s As String In _validatorMessages.WarningMessages
            msgString += s & vbCrLf
        Next
        Return msgString
    End Function

#End Region

#Region "Functions"

    Public Sub New()
        _validatorMessages = New ValidatorMessages
    End Sub

    Public Sub AddCriticalMessage(ByVal message As String)
        _validatorMessages.CriticalMessages.Add(message)
    End Sub

    Public Sub AddWarningMessage(ByVal message As String)
        _validatorMessages.WarningMessages.Add(message)
    End Sub

#End Region

End Class

Public Class ValidationException : Inherits Exception

#Region "Properties"

    Private m_valid As New BaseValidator
    Public Property Validator() As BaseValidator
        Get
            Return m_valid
        End Get
        Set(ByVal Value As BaseValidator)
            m_valid = Value
        End Set
    End Property

#End Region

#Region "Functions"

    Public Sub New(ByVal inValidator As BaseValidator)
        m_valid = inValidator
    End Sub

#End Region

End Class